tests/python/rest_api/test_tus_uploads.py

```diff
@@ -0,0 +1,254 @@
+        task = self._create_task_via_api(
+            {
+                "name": "test TUS resume interrupted upload",
+                "labels": [{"name": "car"}],
+            }
+        )
+
+        image_file = generate_image_file("test_image.jpg", size=(200, 200))
+        image_data = image_file.getvalue()
+        file_size = len(image_data)
+        first_chunk_size = int(file_size * 0.4)
+
+        file_id = self._start_upload(task.id, file_size, "test_image.jpg")
+
+        # Upload 40% of file
+        chunk1 = image_data[:first_chunk_size]
+        response = self._upload_chunk(task.id, file_id, 0, chunk1)
+        assert response.status_code == HTTPStatus.NO_CONTENT
+        assert int(response.headers.get("Upload-Offset")) == first_chunk_size
+
+        # Simulate interruption - check current offset
+        current_offset = self._get_upload_offset(task.id, file_id)
+        assert current_offset == first_chunk_size
+
+        # Resume from where we left off
+        chunk2 = image_data[first_chunk_size:]
+        response = self._upload_chunk(task.id, file_id, current_offset, chunk2)
+        assert response.status_code == HTTPStatus.NO_CONTENT
+        assert int(response.headers.get("Upload-Offset")) == file_size
+
+    def test_upload_offset_is_updated_incrementally(self):
+        """Test offset updates correctly after each small chunk"""
+        task = self._create_task_via_api(
+            {
+                "name": "test TUS incremental offset",
+                "labels": [{"name": "car"}],
+            }
+        )
+
+        image_file = generate_image_file("test_image.jpg", size=(150, 150))
+        image_data = image_file.getvalue()
+        file_size = len(image_data)
+
+        file_id = self._start_upload(task.id, file_size, "test_image.jpg")
+
+        chunk_size = 1024
+        offset = 0
+
+        while offset < file_size:
+            chunk_end = min(offset + chunk_size, file_size)
+            chunk = image_data[offset:chunk_end]
+
+            response = self._upload_chunk(task.id, file_id, offset, chunk)
+            assert response.status_code == HTTPStatus.NO_CONTENT
+
+            new_offset = int(response.headers.get("Upload-Offset"))
+            assert new_offset == chunk_end
+
+            # Verify via HEAD only if not complete (file gets cleaned after completion)
+            if new_offset < file_size:
+                head_offset = self._get_upload_offset(task.id, file_id)
+                assert head_offset == new_offset
+
+            offset = new_offset
+
+    def test_cannot_upload_with_wrong_offset(self):
+        """Test that upload fails if offset doesn't match"""
+        task = self._create_task_via_api(
+            {
+                "name": "test TUS wrong offset",
+                "labels": [{"name": "car"}],
+            }
+        )
+
+        # Generate larger image to ensure multiple chunks needed
+        image_file = generate_image_file("test_image.jpg", size=(200, 200))
```

Comment:
Given that you never complete the upload in this test, there's no reason to generate valid data. You can just generate a string of zeros, which is faster and guarantees that it has the correct length.